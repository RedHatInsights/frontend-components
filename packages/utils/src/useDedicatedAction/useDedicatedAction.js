import React from 'react';

// TODO Refactor.
const useDedicatedAction = (options) => {
  const enableDedicatedAction = !!options.dedicatedAction;
  const { dedicatedAction: DedicatedActionOption, AdditionalDedicatedActions, selected } = options;

  return enableDedicatedAction
    ? {
        toolbarProps: {
          dedicatedAction: (
            <div>
              {DedicatedActionOption && <DedicatedActionOption {...(selected && { selected })} />}
              {AdditionalDedicatedActions && <AdditionalDedicatedActions {...(selected && { selected })} />}
            </div>
          ),
        },
      }
    : {};
};

export default useDedicatedAction;
