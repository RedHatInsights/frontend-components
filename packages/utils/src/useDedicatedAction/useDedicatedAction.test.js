import React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import useDedicatedAction from './useDedicatedAction';

describe('useDedicatedAction', () => {
  it('returns a dedicated action toolbar config', () => {
    const { result } = renderHook(() =>
      useDedicatedAction({
            dedicatedAction: () => ( //eslint-disable-line
          <span>DEDICATED ACTION</span>
        ),
            additionalDedicatedActions: [() => ( //eslint-disable-line
            <span>ANOTHER DEDICATED ACTION</span>
          ),
        ],
        selected: [],
      })
    );
    expect(result).toMatchSnapshot();
  });
});
