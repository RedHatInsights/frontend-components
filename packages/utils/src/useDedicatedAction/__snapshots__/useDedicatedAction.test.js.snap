// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDedicatedAction returns a dedicated action toolbar config 1`] = `
Object {
  "all": Array [
    Object {
      "toolbarProps": Object {
        "dedicatedAction": <div>
          <dedicatedAction
            selected={Array []}
          />
        </div>,
      },
    },
  ],
  "current": Object {
    "toolbarProps": Object {
      "dedicatedAction": <div>
        <dedicatedAction
          selected={Array []}
        />
      </div>,
    },
  },
  "error": undefined,
}
`;
